data_model/src/query/predicate.rs

```diff
@@ -688,9 +689,10 @@ pub mod string {
                 // TODO: Once #2302 and #1889 are merged, add more cases.
             }
 
+            #[cfg(compile_err)] // FIXME the trait `Alias<AccountId>` is not implemented for `str`
             #[test]
             fn account_id() {
-                let id = IdBox::AccountId("alice@wonderland".parse().expect("Valid"));
+                let id = IdBox::AccountId("alice@wonderland".parse_alias());
```

Comment:
As an unresolved issue, 5 tests are failing to build and compiler says

> the trait `Alias<AccountId>` is not implemented for `str`

even though the trait `Alias<AccountId>` is implemented for `str`

Reply:
That's strange everything compiles just fine on my machine.

Reply:
Implementing for `&str` did not work. I guess it has something to do with `http` and `transparent_api` features

Reply:
if we disallow strings that parse as multihash to be used as aliases then instead of having separate `parse_alias` method we can put all the logic into `FromStr::from_str`.

Reply:
In that case,

- FromStr defined in the data model depends on the sample aliases
- deserializer accepts both multihash and alias, but serializer must decide between them

Do we accept them?

Reply:
> FromStr defined in the data model depends on the sample aliases

I don't understand why

Reply:
I assume your suggestion would be something like:

```rust
// data_model/src/account.rs
impl FromStr for AccountId {
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        if is_multihash(s) {
            todo!()
        } else {
            use iroha_sample_params::alias::Alias;
            let account_id = s.parse_alias(); // or something equivalent inlined
            Ok(account_id)
        }
    }
}

```

If so, it would be referencing to something like the current `configs/sample_params/src/.toml`, which is only for testing

Reply:
you don't have to put your code into `Alias` trait just paste it here

Reply:
I mean, going into the `else` block would not be expected except in test cases.
I assume client should have its own alias resolution before building transaction and should communicate with peer in resolved format, where `FromStr` implementation plays an role of the interface.
So for non-multihash (unresolved) input `from_str` should return `Err` immediately

Reply:
>  I assume client should have its own alias resolution before building transaction and should communicate with peer in resolved format, where `FromStr` implementation plays an role of the interface.

ok, I understand you now but I don't like it. I don't want to see 2 different methods for parsing account ids. User shouldn't worry which format is his account id in. And then if they change the format their whole code would break.
I'd rather we put this into `data_model` and introduce some feature flag to control this behavior. We can actually reuse `transparent_api` here. For clients `account_id.parse()` will work for both aliases and keys, but in `iroha_core` it will return error for aliases. How about this?

Reply:
I still don't understand your point. What's the problem with the following?
- user (end user with client) is allowed to enter either multihash or alias
  - the client should have a different resolver (name service) other than `parse_alias`
    - probably try to `parse` as multihash first and then ask the name service to resolve alias
- user (developer) will be writing tests with `parse_alias` in most cases, as long as our integration tests rely on it
- if any fragility with `parse_alias`, the same with that consolidated `parse`

Reply:
Isn't your suggestion to switch resolver depending on whether it is testing environment or not, rather than called from clients or not?

Reply:
how about if we used only public key format of account ids in `data_model`.  You can keep `parse_alias` in `iroha_client`

Reply:
I think `AccountId` in data model is already restricted to the public key (multihash) format as shown [here](https://github.com/hyperledger/iroha/pull/4411#discussion_r1552265003)

Reply:
yes, but in `data_model` tests you use `parse_alias`

Reply:
Yeah, currently `iroha_sample_params` appears in
- `[dev-dependencies]` of core, data_model, data_model_derive, genesis
- `[dependencies]` of client, test_network, kagami, parity_scale_cli
Is it your goal to remove dev dependency on `iroha_sample_params` from data_model?

Also, I noticed even if we keep this crate it should be renamed to drop `iroha_` prefix as it is not planed to publish

Reply:
> Is it your goal to remove dev dependency on iroha_sample_params from data_model?

no, but it's a nice consequence if it happens. My goal is to not introduce concept of aliases into `data_model`

Reply:
Ok, then I'll just replace alias with multihash in data_model. What do you think about other crate?

Reply:
I'd also replace aliases with public keys in other crates

Reply:
I also don't particularly like the idea of having a global `.parse_alias` extension, in `data_model` or not. However, having a pool of aliases for testing purposes seems useful.

A middle ground, IMO, would be having a utility struct like `test_aliases::TestAliases` which will provide a simple and expressive API. For example:

```rust
// voila: 
let alice: AccountId = TestAliases::get("alice").in_domain("wonderland");
```

Or even simpler, just returning a `PublicKey` for a specific account name.

Reply:
> However, having a pool of aliases for testing purposes seems useful.

I disagree. Since the value doesn't matter, just create a new data where you need it. Sharing test data across crates looks crazy to me

Reply:
On second thought, the only purpose is to easily get an `AccountId` for testing.
The word "alias" might not be good either.
No need to fix keys except for a few predefined signatories: peer, genesis, alice in test network and bob, carpenter in default genesis

Reply:
Imo having aliases in integration tests is convenient feature.

Reply:
> I disagree. Since the value doesn't matter, just create a new data where you need it. Sharing test data across crates looks like an antipattern to me

That point works for me. Indeed, why don't generate random `AccountId`s in tests each time?

Reply:
Considering that account names were originally hard-coded in tests, even fixed keys would serve as a refactoring.
Btw pytests do not require any predefined signatories so far

Reply:
> Here is genesis though for example, so some way (the current toml file would be a candidate) to share predefined signatories across languages would be desired when such tests become not ignored

wdym by share across languages? can you be more specific which crates need to share predefined signatories

Reply:
Ok, random keys might work just as good, i won't insist on aliases.

Reply:
Ok, I think this discussion is settling down:
I'll withdraw the whole `iroha_sample_params`. Instead, since `test_network` will be deprecated and unit tests are still generating account IDs, I'm going to introduce an independent utility crate `test_samples`, which provides some generating function like:

```rust
use test_samples::gen_account_in;

let (mouse_id, mouse_keypair) = gen_account_in("wonderland");
```
As for alice, I might have trouble dealing with it